General improvements and fixes (#16)
woojiahao/git-guide: CONTRIBUTING.md

## Self-hosting

> To host the guide on GitHub Pages, the quickest way is by enabling it in the settings of GitHub. Set GitHub Pages to render from the `gh-pages` branch (refer [here](https://docs.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)). Once this is done, GitHub Pages will take a while to upload your changes and it will soon be available at `https://<username>.github.io/git-guide`.

Author: `gh-pages` is the branch that hosts the website, but how do we get github to generate the static website like u did from publish? (I am assuming the `publish` branch has some workflow to build and push the static webpage to `gh-pages`)  

Do we need to add this info in?

Reviewer: We don't need to include this for the CONTRIBUTING.md. The way the page deploys isn't super important but if you want to include it, I have a blog post detailing how the entire things works already: https://woojiahao.github.io/blog/posts/2020-04-14-Gatsby.js%20+%20GitHub%20Actions%20+%20GitHub%20Pages%20=%20Match%20Made%20in%20Heaven.

Reviewer: Though, I'd say that this probably goes better in the `site/`'s README more than CONTRIBUTING.md

Author: Yea I was thinking about that as well because I don't think without building the site, it will show up on `gh-pages`.  
~~I might be wrong on this, let me check~~

Just checked, yea I guess if we are talking about hosting the `gh-pages` branch on your own.

## Self-hosting

> ~~To host the guide on GitHub Pages, the quickest way is by enabling it in the settings of GitHub. Set GitHub Pages to render from the `master` branch (refer [here](https://docs.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)). Once this is done, GitHub Pages will take a while to upload your changes and it will soon be available at `https://<username>.github.io/git-guide`.~~

Reviewer: ```suggestion
To host the guide on GitHub Pages, the quickest way is by enabling it in the settings of GitHub. Set GitHub Pages to render from the `gh-pages` branch (refer [here](https://docs.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)). Once this is done, GitHub Pages will take a while to upload your changes and it will soon be available at `https://<username>.github.io/git-guide`.
```

> ~~**Note*** it is possible to use a separate documentation generator for this but for simplicity sake, we will be using the default GitHub Pages template.~~

Reviewer: ```suggestion
**Note*** it is possible to use a separate documentation generator for this but for simplicity sake, we will be using the default GitHub Pages template.
```

Author: Actually I feel this this should be removed since it is using Gatbsy.js.  
But if you want to leave it in, I fine with it too.

> Update as of **01 August 2020** we are now using Gatsby.js to host our guide on github pages.  

Reviewer: ```suggestion
As of **01 August 2020** we are now using Gatsby.js to generate our guide on GitHub pages.
```

> You can find out more information on how to self-host the website [here.](./site/README.md)

Author: Not sure to leave in the old self-host instructions

Reviewer: They would actually have to enable this option in the settings regardless of the self-host option. That's a prerequisite to actually viewing the site on GitHub pages.

Author: Ah I see, I thought the development set up was how you could see the website locally.  
I guess you would still need the `gh-pages` branch to self-host it.
